middlewares/auth.js

```diff
@@ -1,6 +1,35 @@
-export const auth = (req, res, next) => {
-    // TODO: implement the auth middleware here
+import jwt from "jsonwebtoken";
+import { findUserById } from "../services/user/index.js";
+import { ApiError } from "../utils/ApiError.js";
 
-    
+/**
+ * Middleware for user authentication using JWT token
+ * @param {Object} req - Express request object
+ * @param {Object} res - Express response object
+ * @param {Function} next - Express next function
+ */
+
+export const auth = async (req, res, next) => {
+  // TODO: implement the auth middleware here
+
+  try {
+    const token = req.header("Authorization")?.replace("Bearer ", "");
+
+    if (!token) {
+      throw new ApiError(401, "Unauthorized request");
+    }
+
+    const decodedToken = jwt.verify(token, process.env.JWT_SECRET);
+
+    const user = await findUserById(decodedToken?.userId);
+
+    if (!user) {
+      throw new ApiError(401, "Invalid Access Token");
+    }
+
+    req.user = user;
```

Comment:
adding user objects directly from the DB exposes a lot of sensitive data that is not required, such as passwords and others. Let's assign the user id to `req.user`